File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    answer = []
    
    # 갈색블럭의 개수는 2(n + m) / 2
    # 노란색은 n * m
    # 가로 길이는 세로 길이와 같거나 길다
    nm_range = int((brown - 4) / 2)
    
    for i in range(nm_range):
        if i * (nm_range - i) == yellow:
            answer.append(i + 2)
            answer.append(nm_range - i + 2)
            break
    
    
    return sorted(answer, reverse=True)

File: 프로그래머스/2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = new int[2];
        
        for (int i = 1; i <= yellow / 2 + 1; i++) {
            if (yellow % i == 0) {
                int x = i;
                int y = yellow / i;
                
                int circum = (2 * x + 2 * y + 4);

                if (circum == brown) {
                    answer[1] = x + 2;
                    answer[0] = y + 2;
                    break;
                }
            }
        }
        
        return answer;
    }
}